elasticsearch/src/main/java/com/tcdi/zombodb/query_parser/QueryRewriter.java

```diff
@@ -379,33 +380,76 @@ private AggregationBuilder build(ASTTally agg) {
         }
     }
 
+    /**
+     * Determine if a particular field name is present in the index
+     *
+     * @param md index metadata
+     * @param fieldname field name to check for
+     * @return true if this field exists, false otherwise
+     */
+    private boolean hasDate(final IndexMetadata md, final String fieldname) {
+        return md.hasField(fieldname + DateSuffix);
+    }
+
     static class RangeSpecEntry {
         public String key;
         public Double from;
         public Double to;
     }
 
+    /**
+     * Container for date range aggregation spec
+     */
+    static class DateRangeSpecEntry {
```

Comment:
So it seems to me that the existing `RangeSpecEntry` and the new `DateRangeSpecEntry` can just be the same class, where "from" and "to" are `java.lang.Object`.

Once that happens, I think the code paths for building a generic RangeBuilder and DateRangeBuilder can be merged into a single code path.  Both the `RangeBuilder` and `DateRangeBuilder` classes inherit from `AbstractRangeBuilder`, so it should be straight forward.